src/vmm/src/guest_config/x86_64/mod.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+use crate::guest_config::cpuid::Cpuid;
+
+/// CPU configuration for x86_64 CPUs
+pub struct X86_64CpuConfiguration {
+    /// CPUID configuration
```

Comment:
If this is a X86 cpu config struct then where is `smt`?

Reply:
So for the user to enable smt he would need to use cpu template? Well, maybe we can have a static template that does that, but seems inconvenient to me.

Reply:
`VcpuConfig` has `smt` field.
https://github.com/firecracker-microvm/firecracker/blob/main/src/vmm/src/vstate/vcpu/mod.rs#L77

I'm not sure what if `VcpuConfig` and CPU template conflicts on SMT.

Reply:
@zulinx86 Yep this is an important point. For that particular case, the custom template will override.